app/models/product_import/product_importer.rb

```diff
@@ -237,6 +237,15 @@ def rows
                                      error_message: e.message))
       end
       []
+    rescue CSV::MalformedCSVError => e
+      # NOTE the init_product_importer method calls build_entries and
```

Comment:
can you please remove NOTE here as well?
just
`# the init_product_importer method calls build_entries and`

Reply:
A rule for code comments I love is: if you need to comment a line of code, extract method and make it a method comment. In this case:
```ruby
  rescue CSV::MalformedCSVError => e
     add_malformed_csv_error(e.message)
     []
  end

  # This error is raised twice because init_product_importer calls both build_entries and buils_all_entries
  def add_malformed_csv_error(error_message)
      return if errors.added?(:importer, I18n.t('admin.product_import.model.malformed_csv', error_message: error_message))

      errors.add(:importer, I18n.t('admin.product_import.model.malformed_csv',error_message: error_message))
  end
```       

I think this looks nicer, what do you think?

Reply:
this change will also help with the codeclimate warning:
Assignment Branch Condition size for rows is too high. [<2, 21, 6> 21.93/15]

Reply:
> A rule for code comments I love is: if you need to comment a line of code, extract method and make it a method comment. 

Make sense, although I feel like It's just adding a method to comply with the codeclimate rules. Anyway it does make the code a bit more readable.

Rubocop still throw a warning with the suggested changes : 
`app/models/product_import/product_importer.rb:226:5: C: Metrics/AbcSize: Assignment Branch Condition size for rows is too high. [<2, 15, 5> 15.94/15]`

I am not sure if there is a way to make it better, what do you think ?

Reply:
>>  although I feel like It's just adding a method to comply with the codeclimate rules.

imo code climate rules are very important and readability is fundamental. It's adding a method to make the code a lot more readable and less complex. The complexity rule is very important. Messy code will create bugs and head aches.

The warning is there but now it's 15.94 , not 21. It means it's less complex.
I think we can leave it like that :+1: